File: external_apps/mobile_android/app/src/main/java/com/example/mobile_android/data/profile/AvatarManager.kt
package com.example.mobile_android.data.profile

import android.net.Uri
import com.example.mobile_android.data.media.ImageProcessor
import javax.inject.Inject
import javax.inject.Singleton
import java.io.File

@Singleton
class AvatarManager @Inject constructor(
    private val imageProcessor: ImageProcessor
) {
    /**
     * Process avatar image from URI
     * Compresses, crops to square, and prepares for upload
     */
    suspend fun processAvatar(uri: Uri): File? {
        // Compress and resize to reasonable avatar size (e.g., 512x512)
        val processedFile = imageProcessor.compressImage(
            uri = uri,
            quality = 85,
            maxWidth = 512,
            maxHeight = 512
        )
        
        // In a real app, we might want to enforce cropping here if not done by picker
        return processedFile
    }
    
    /**
     * Upload avatar
     * This would typically call a Repository/API
     */
    suspend fun uploadAvatar(file: File): Result<String> {
        // Placeholder for API upload call
        return try {
            kotlinx.coroutines.delay(1000)
            Result.success("https://example.com/avatars/${file.name}")
        } catch (e: Exception) {
            Result.failure(e)
        }
    }
}

File: external_apps/mobile_ios/mobile_ios/Core/Profile/AvatarManager.swift
//
//  AvatarManager.swift
//  mobile_ios
//
//  Created for Profile Feature
//

import UIKit

class AvatarManager {
    static let shared = AvatarManager()
    
    private let imageProcessor = ImageProcessor.shared
    
    private init() {}
    
    /// Process avatar image
    /// Compresses and prepares for upload
    func processAvatar(image: UIImage, completion: @escaping (Result<Data, Error>) -> Void) {
        // Compress to reasonable size
        if let compressedData = imageProcessor.compressImage(image, quality: 0.8) {
            completion(.success(compressedData))
        } else {
            completion(.failure(NSError(domain: "AvatarManager", code: -1, userInfo: [NSLocalizedDescriptionKey: "Failed to compress image"])))
        }
    }
    
    /// Upload avatar
    func uploadAvatar(data: Data, completion: @escaping (Result<String, Error>) -> Void) {
        // Placeholder for API upload call
        DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
            completion(.success("https://example.com/avatars/avatar_\(UUID().uuidString).jpg"))
        }
    }
}
